autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgentTest.java

```diff
@@ -0,0 +1,665 @@
+    }
+
+    @Test
+    public void getClusterToManageTestEmptyEnabledZonesList() {
+        Mockito.doReturn(new ArrayList<>()).when(zoneService).listAllZonesEnabled();
+
+        ClusterVO result = administrationAgent.getClusterToManage(clusterAdministrationHeuristicAlgorithm);
+        Assert.assertNull(result);
+        verifyGetClusterToManageCalledMethods(0);
+    }
+
+    @Test
+    public void getClusterToManageTestNotEmptyEnabledZonesList() {
+        List<DataCenterVO> zones = new ArrayList<>();
+        zones.add(new DataCenterVO(0, "name", "description", "", "", "", "", "", "", 0l, NetworkType.Basic, "", ""));
+        ClusterVO mockedCluster = Mockito.mock(ClusterVO.class);
+
+        Mockito.doReturn(zones).when(zoneService).listAllZonesEnabled();
+        Mockito.doReturn(mockedCluster).when(administrationAgent).getClusterToManageFromZone(Mockito.any(ClusterAdministrationHeuristicAlgorithm.class), Mockito.anyLong());
+
+        ClusterVO result = administrationAgent.getClusterToManage(clusterAdministrationHeuristicAlgorithm);
+
+        Assert.assertEquals(mockedCluster, result);
+        Assert.assertNotNull(result);
+        verifyGetClusterToManageCalledMethods(1);
+    }
+
+    @Test
+    public void getClusterToManageFromZoneTestListOfClustersEmpty() {
+        List<ClusterVO> clusters = new ArrayList<>();
+        Mockito.doReturn(clusters).when(clusterService).listAllClustersOnZone(Mockito.anyLong());
+
+        administrationAgent.getClusterToManageFromZone(clusterAdministrationHeuristicAlgorithm, null);
+
+        verifyGetClusterToManageFromZoneCalledMethods(0, 0, 0);
+    }
+
+    @Test
+    public void getClusterToManageFromZoneTestListOfClustersNotEmptyAndClusterIsNotAdministrated() {
+        List<ClusterVO> clusters = createClusters(null);
+        getClusterToManageFromZoneReturnedValuesSetup(clusters, false, true);
+
+        administrationAgent.getClusterToManageFromZone(clusterAdministrationHeuristicAlgorithm, null);
+
+        verifyGetClusterToManageFromZoneCalledMethods(1, 1, 1);
+    }
+
+    @Test
+    public void getClusterToManageFromZoneTestListOfClustersNotEmptyAndCannotProcess() {
+        List<ClusterVO> clusters = createClusters(null);
+        getClusterToManageFromZoneReturnedValuesSetup(clusters, false, false);
+
+        administrationAgent.getClusterToManageFromZone(clusterAdministrationHeuristicAlgorithm, null);
+
+        verifyGetClusterToManageFromZoneCalledMethods(1, 1, 0);
+    }
+
+    @Test
+    public void getClusterToManageFromZoneTestClusterIsRemoved() {
+        List<ClusterVO> clusters = createClusters(new Date());
+        getClusterToManageFromZoneReturnedValuesSetup(clusters, true, true);
+
+        administrationAgent.getClusterToManageFromZone(clusterAdministrationHeuristicAlgorithm, null);
+
+        verifyGetClusterToManageFromZoneCalledMethods(0, 0, 0);
+    }
+
+    @Test
+    public void workOnClusterTest() {
```

Comment:
what flow of execution is being tested here?
this method has at least two, the happy day, and another happening an exception when executing "processCluster"